robottelo/cli/factory.py

```diff
@@ -892,6 +901,10 @@ def make_registry(options=None):
 
 @cacheable
 def make_repository(options=None):
+    return make_product_with_credentials(options)
+
+
+def make_repository_with_credentials(options=None, credentials=None):
```

Comment:
Think we can refactor the current function by adding credentials kwarg with default None value, without creating additional functions, what do you think?
```python
def make_repository(options=None, credentials=None):
    ...
```

Reply:
That is what I tried at first. But that way @cacheable is designed prevent it since. Check it has options and cache as boolean. So it would be kind of hard doing that right now since a lot of test are using this function. But anyway this is an example of complexity that I mentioned when proposed to start a new CLI approach in parallel with current one ;)